### ISSUE TITLE
Mobile view of dashboard displays a disorganized toolbar

### ISSUE THREAD

**Issue by jeff-bruemmer** [metabase/metabase#17354]
**Describe the bug**
The toolbar on a dashboard, especially the collection name, needs some attention (see screenshot).

**Logs**
This is just a styling issue.

**To Reproduce**
1. On a phone, or in developer tools in a phone view, open a dashboard.

**Expected behavior**
A legible dashboard toolbar, with responsive layout. 

**Screenshots**
<img width="464" alt="mobile-view-of-dashboard" src="https://user-images.githubusercontent.com/17343328/128517740-fe030194-a160-4400-a0e5-caacaa1789d5.png">

**Information about your Metabase Installation:**
```
{
  "browser-info": {
    "language": "en-US",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.107 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.9.1+1",
    "java.vendor": "AdoptOpenJDK",
    "java.vendor.url": "https://adoptopenjdk.net/",
    "java.version": "11.0.9.1",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.9.1+1",
    "os.name": "Mac OS X",
    "os.version": "10.16",
    "user.language": "en",
    "user.timezone": "America/New_York"
  },
  "metabase-info": {
    "databases": [
      "h2"
    ],
    "hosting-env": "unknown",
    "application-database": "h2",
    "application-database-details": {
      "database": {
        "name": "H2",
        "version": "1.4.197 (2018-03-18)"
      },
      "jdbc-driver": {
        "name": "H2 JDBC Driver",
        "version": "1.4.197 (2018-03-18)"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2021-08-05",
      "tag": "vUNKNOWN",
      "branch": "master",
      "hash": "988ae66"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

**Severity**
Cosmetic. I doubt people will be editing dashboards on their phones.

**Comment by flamber** [comment]
Regression on master - this is 0.40.2, which could use some tweaking, but is okay:
![image](https://user-images.githubusercontent.com/1447303/128519457-4e992d98-ffb2-442e-b29b-eaa88c3d388b.png)

**Comment by kulyk** [comment]
Fixed by #21349
